KeenClient/KIODBStore.m

```diff
@@ -982,85 +990,89 @@ - (BOOL)checkOpenDB:(NSString *)failureMessage {
     return true;
 }
 
-- (void)prepareSQLStatement:(keen_io_sqlite3_stmt **)sqlStatement sqlQuery:(char *)sqlQuery failureMessage:(NSString *)failureMessage {
+- (BOOL)prepareSQLStatement:(keen_io_sqlite3_stmt **)sqlStatement sqlQuery:(char *)sqlQuery failureMessage:(NSString *)failureMessage {
     if(keen_io_sqlite3_prepare_v2(keen_dbname, sqlQuery, -1, sqlStatement, NULL) != SQLITE_OK) {
         [self handleSQLiteFailure:failureMessage];
+        return NO;
     }
+    return YES;
 }
 
 - (void)resetSQLiteStatement:(keen_io_sqlite3_stmt *)sqliteStatement {
     keen_io_sqlite3_reset(sqliteStatement);
     keen_io_sqlite3_clear_bindings(sqliteStatement);
 }
 
-- (void)prepareAllSQLiteStatements {
+- (BOOL)prepareAllSQLiteStatements {
 
     // EVENT STATEMENTS
     
     // This statement inserts events into the table.
-    [self prepareSQLStatement:&insert_event_stmt sqlQuery:"INSERT INTO events (projectID, collection, eventData, pending, attempts) VALUES (?, ?, ?, 0, 0)" failureMessage:@"prepare insert event statement"];
+    if(![self prepareSQLStatement:&insert_event_stmt sqlQuery:"INSERT INTO events (projectID, collection, eventData, pending, attempts) VALUES (?, ?, ?, 0, 0)" failureMessage:@"prepare insert event statement"]) return NO;
```

Comment:
This is awfully verbose, but there's really not a cleaner way to do it without refactoring the entire method.